src/app/shared/navigation-focus/navigation-focus.ts

```diff
@@ -9,16 +11,32 @@ let lastTimeoutId = -1;
 export class NavigationFocus implements OnInit {
   @HostBinding('tabindex') role = '-1';
 
-  constructor(private el: ElementRef) {}
+  constructor(private el: ElementRef, private router: Router) {}
 
   ngOnInit() {
-    clearTimeout(lastTimeoutId);
-    // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
-    lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus({preventScroll: true}),
-      100);
+    let previousUrl = window.location.hostname;
+    this.router.events
+      .pipe(filter((event: Event) => event instanceof NavigationEnd))
+      .subscribe(() => {
+        const afterUrl = window.location.hostname;
+        // only want to focus on the element if it the navigation was a soft nav,
+        // otherwise first focusable element should be the skip link
+        if (isSoftNav(previousUrl)) {
+          // 100ms timeout is used to allow the page to settle before moving focus for screen readers.
+          clearTimeout(lastTimeoutId);
+          lastTimeoutId = window.setTimeout(() => this.el.nativeElement.focus({preventScroll: true}),
+            100);
+        }
+        previousUrl = afterUrl;
+      });
   }
 }
 
+function isSoftNav(url: string) {
+  const withinApp = /(localhost|material.angular.io)/;
```

Comment:
This is brittle, I would just compare the old and new hostname with `===`.

@atscott do you know if there's a more "Angular" way of distinguishing soft nav from hard nav?

Reply:
We'll want this to work with the following as well
- https://next.material.angular.io/
- https://v6.material.angular.io/ (also v5, v7, v8, etc)